nexus/tests/integration_tests/unauthorized.rs

```diff
@@ -55,22 +56,29 @@ async fn test_unauthorized(cptestctx: &ControlPlaneTestContext) {
     DiskTest::new(cptestctx).await;
     let client = &cptestctx.external_client;
     let log = &cptestctx.logctx.log;
+    let mut setup_results = std::collections::BTreeMap::new();
 
     // Create test data.
     info!(log, "setting up resource hierarchy");
     for request in &*SETUP_REQUESTS {
-        NexusRequest::objects_post(client, request.url, &request.body)
-            .authn_as(AuthnMode::PrivilegedUser)
-            .execute()
-            .await
-            .unwrap();
+        let result =
+            NexusRequest::objects_post(client, request.url, &request.body)
+                .authn_as(AuthnMode::PrivilegedUser)
+                .execute()
+                .await
+                .unwrap();
+
+        setup_results.insert(request.url, result.clone());
+        request.id_routes.iter().for_each(|id_route| {
+            setup_results.insert(id_route, result.clone());
+        });
     }
 
     // Verify the hardcoded endpoints.
     info!(log, "verifying endpoints");
     print!("{}", VERIFY_HEADER);
     for endpoint in &*VERIFY_ENDPOINTS {
-        verify_endpoint(&log, client, endpoint).await;
+        verify_endpoint(&log, client, endpoint, &setup_results).await;
```

Comment:
I'd rather do the lookup of the result here and pass that in vs. passing in the entire list